ambari-server/src/main/java/org/apache/ambari/server/controller/internal/UpgradeResourceProvider.java

```diff
@@ -857,6 +857,8 @@ major stack versions (e.g., HDP 2.2 -> 2.3), and then set config changes
               Map<String, String> requestProperties = new HashMap<>();
               requestProperties.put(SupportedCustomOperation.REGENERATE_KEYTABS.name().toLowerCase(), "missing");
               requestProperties.put(KerberosHelper.ALLOW_RETRY, Boolean.TRUE.toString().toLowerCase());
+              requestProperties.put(KerberosHelper.DIRECTIVE_IGNORE_CONFIGS,
```

Comment:
Unfortunately, this will prevent any changes to the configurations related to new Kerberos identities, which will cause failures if new configurations are expected.

Reply:
I think a better change would be to remove any static configuration changes from the Kerberos descriptor, add them to service advisors, getServiceConfigurationRecommendationsForKerberos method for the relevant services.